Android/app/src/main/java/app/intra/sys/firebase/MobileCountryCode.java

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2019 Jigsaw Operations LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app.intra.sys.firebase;
+
+import android.content.Context;
+import android.telephony.TelephonyManager;
+import java.lang.reflect.Method;
+
+/**
+ * Static utility function for getting the Mobile Country Code.
+ */
+class MobileCountryCode {
+  // Mobile Country Code, or 0 if it could not be determined.  This is used to memoize get()
+  // across all the analytics events.
+  private static Integer mcc = null;
```

Comment:
Let's move the memoization to the AnalyticsWrapper instead, since this is higher level logic that can cause confusion at a low level (really an application optimization)

The logic of only using the country if they match could go in the AnalyticsWrapper too, since that's clearly application logic.

Reply:
Done